### PR TITLE
alShaders part 1

### PR DESCRIPTION
**Pull request by est77**
- alSurface layer BRDF fixes and improvements.
- Imported OSL utility shaders from alShaders.
- Started alSurface shader.
- Use the same inputs for the Gaussian and the other BSSRDFs.
- Expose the Gaussian BSSRDF to OSL.
- Add a fresnel weight input to BSSRDFs.
- Maya area light shader improvements.

### CODE REVIEW

**Comment by dictoon** [comment]
Looks great!

Just one thing: please add a project update step to remap `v` to `radius` for the Gaussian BSSRDF in older projects.

File: src/appleseed/renderer/modeling/bssrdf/gaussianbssrdf.cpp [hunk 1]
```diff
@@ -178,7 +178,7 @@ namespace
             for (size_t i = 0, e = values->m_precomputed.m_channel_pdf.size(); i < e; ++i)
             {
                 // The remapping from mfp to radius comes from alSurface.
-                const float radius = values->m_mfp[i] * 7.0f;
+                const float radius = max(values->m_mfp[i], 0.0001f) * 7.0f;
```
Comment (dictoon) [inline src/appleseed/renderer/modeling/bssrdf/gaussianbssrdf.cpp]: Please include `<algorithm>`.

File: src/appleseed/renderer/modeling/project/projectfileupdater.cpp [hunk 2]
```diff
@@ -1245,6 +1249,43 @@ namespace
...
+            {
+                const float v = params.get<float>("v");
+                const float mfp = sqrt(v * 16.0f) / 7.0f;
+                params.insert("mfp", mfp);
+                params.remove_path("v");
+            }
+            catch (const Exception&)
+            {
+                RENDERER_LOG_ERROR(
+                    "while updating gaussianbssrdf \"%s\", failed to convert v parameter.",
```
Comment (dictoon) [inline src/appleseed/renderer/modeling/project/projectfileupdater.cpp]: *gaussian bssrdf* maybe?

**Comment by est77** [comment]
I'll include the fixes in the next PR.

**Pull request merged**